apps/web/vibes/soul/components/cart/index.tsx

```diff
@@ -1,231 +1,226 @@
-  useEffect(() => {
-    const timer = setTimeout(() => {
-      setIsLoading(false)
-    }, 3000)
-    return () => {
-      clearTimeout(timer)
-    }
-  }, [])
+async function CartUI({
+  title,
+  lineItems,
+  summary,
+  emptyState,
+  removeLineItemAction,
+  updateLineItemQuantityAction,
+  // redirectToCheckoutAction,
+}: CartProps) {
+  const resolvedLineItems = await Promise.resolve(lineItems)
+
+  const calculateCartQuantity = (items: CartLineItem[]) => {
+    return items.reduce((total, item) => total + item.quantity, 0)
+  }
 
   return (
     <div className="mx-auto max-w-screen-2xl @container">
       <div className="flex w-full flex-col gap-10 px-3 pb-10 pt-24 @xl:px-6 @4xl:flex-row @4xl:gap-20 @4xl:pb-20 @4xl:pt-32 @5xl:px-20">
         {/* Cart Side */}
-        <div className={clsx(products.length > 0 && '@4xl:w-2/3', 'w-full')}>
+        <div className={clsx(resolvedLineItems.length > 0 && '@4xl:w-2/3', 'w-full')}>
           <h1 className="mb-10 font-heading text-4xl font-medium leading-none @xl:text-5xl">
-            Your Cart
-            {!isLoading && products.length > 0 && (
-              <span className="ml-4 text-contrast-200">{products.length}</span>
+            {title}
+            {resolvedLineItems.length > 0 && (
+              <span className="ml-4 text-contrast-200">
+                {calculateCartQuantity(resolvedLineItems)}
+              </span>
             )}
           </h1>
 
           {/* Cart Items */}
           <ul className="flex flex-col gap-5 gap-y-10">
-            {isLoading ? (
-              <div className="flex animate-pulse items-center gap-x-5">
-                <div className="h-96 w-full rounded-lg bg-contrast-100" />
-              </div>
-            ) : (
-              // Cart Items
-              products.map(({ id, name, href, image, price, subtitle, quantity }) => (
+            {resolvedLineItems.map(
+              ({ id, title: lineItemTitle, image, price, subtitle, quantity }) => (
                 <li
                   className="flex flex-col items-start gap-x-5 gap-y-6 @sm:flex-row @sm:items-center @sm:gap-y-4"
                   key={id}
                 >
-                  {image && (
-                    <Link
-                      href={href}
-                      className="relative aspect-[3/4] w-full overflow-hidden rounded-lg bg-contrast-100 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-primary focus-visible:ring-offset-4 @sm:max-w-36"
-                    >
-                      <Image fill src={image.src} alt={image.altText} className="object-cover" />
-                    </Link>
-                  )}
+                  <div className="relative aspect-[3/4] w-full overflow-hidden rounded-lg bg-contrast-100 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-primary focus-visible:ring-offset-4 @sm:max-w-36">
+                    <Image fill src={image.src} alt={image.altText} className="object-cover" />
```

Comment:
We actually need to pass in a `sizes` prop for this `Image` to prevent grabbing massive images.